devel/floppy/Portfile

```
@@ -0,0 +1,61 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+PortGroup       compilers 1.0
+
+name            floppy
+version         7.0
+revision        0
+categories      devel
+platforms       darwin
+license         Noncommercial
+maintainers     {@kamischi web.de:karl-michael.schindler} openmaintainer
+homepage        http://www.netlib.org/floppy/
+master_sites    ${homepage}
+distfiles       ${name}7.tgz
+dist_subdir     ${name}/${version}
+extract.mkdir   yes
+description     Fortran source code analyzer
+
+long_description {
+FLOPPY is a software tool that takes as input a file of FORTRAN 77 code \
+and checks it according to various "coding conventions". Floppy can \
+"tidy" the source FORTRAN, producing a new file with indented DO-loops, \
+block IF-s, and so on. Floppy can be used to generate HTML from the \
+Fortran program. In this case, a new file is written where each module \
+and include file name in the source Fortran is replaced by an HTML Anchor. \
+The format of the source is preserved when the document is browsed by an \
+HTML browser. FLOPPY can also be used to write a binary summary file \
+which is then used as input to the FLOW program.
+}
+
+checksums   rmd160 c7a84b018df98a7a53edb12481e3e29b7f50ea4e \
+            sha256 f76735b24f76ab3c3d2ca6a276416890676694721bdb57f9c74cde36c548c68e \
+            size   175655
+
+post-patch {
+    # newer fortrans have a function range. Rename it.
+    reinplace -W ${worksrcpath} "s|RANGE|EGNAR|g" range.for procom.for settyp.for
+    # set compiler and additional options
+    reinplace -W ${worksrcpath} "s|f77|${configure.f77} -fdiagnostics-color=always|g" makefile
+}
+
+compilers.choose    f77
+compilers.setup     require_fortran
+use_configure       no
+build.target        floppy
+build.args          -f makefile
+use_parallel_build  no
+
+destroot {
+    xinstall -m 0755 -W ${worksrcpath} floppy \
+        ${destroot}${prefix}/bin
+    xinstall -d ${destroot}${prefix}/share/man/manl
+    xinstall -m 0444 -W ${worksrcpath} floppy.l \
+        ${destroot}${prefix}/share/man/manl
```

For both ports: I've never heard of a manpage category `l` (lowercase L) before. Is this a case of the developer not realizing that the manpage should have been named `floppy.1` (number one) not `floppy.l` (lowercase L) because they look similar? I think so, since the first line of `floppy.l` reads:
```
.TH floppy 1 "3rd March 1993"  "CERN"
```
(with a number one, not a lowercase L).

In both ports, I recommend installing the file with the correct name and in the correct directory (number one, not lowercase L).
```suggestion
    xinstall -m 0444 -W ${worksrcpath} floppy.l \
        ${destroot}${prefix}/share/man/man1/floppy.1
```
In both ports, I suggest adding a patch that fixes the last line of the manpage where it references the manpage of the other port using lowercase L instead of number one, e.g.:
```
--- floppy.l.orig	1994-02-09 02:37:58.000000000 -0600
+++ floppy.l	2020-10-28 08:12:03.000000000 -0500
@@ -278,4 +278,4 @@
 
 .SH SEE ALSO
 .PP
-flow(l), f77(1)
+flow(1), f77(1)
```

---

Good point. I am impressed by your careful inspection. According fix done.